python/ray/serve/scripts.py

```diff
@@ -297,9 +306,15 @@ def run(
         cli_logger.print(f'Deploying from config file: "{config_path}".')
 
         with open(config_path, "r") as config_file:
-            config = ServeApplicationSchema.parse_obj(yaml.safe_load(config_file))
+            config_dict = yaml.safe_load(config_file)
+            # If host or port is specified as a CLI argument, they should take priority
+            # over config values.
+            override_or_default(config_dict, "host", host, DEFAULT_HTTP_HOST)
+            override_or_default(config_dict, "port", port, DEFAULT_HTTP_PORT)
+            config = ServeApplicationSchema.parse_obj(config_dict)
         is_config = True
     else:
+        host, port = host or DEFAULT_HTTP_HOST, port or DEFAULT_HTTP_PORT
```

Comment:
Could we switch this to `if-else` blocks where we explicitly check that the `host` and `port` are `None`? With the current logic, it's impossible to set `port` to 0. Although that should be fine since port 0 is reserved, it's unexpected that it gets overridden quietly.